app/src/main/java/unipd/se18/ocrcamera/PreProcessing.java

```diff
@@ -0,0 +1,369 @@
+
+        //Process the image with the Probabilistic Hough Transform
+        double rho = 1;
+        double theta = Math.PI / 180;
+        int threshold = 50;
+        double minLineLenght = 50;
+        double maxLineGap = 10;
+        Imgproc.HoughLinesP(grayscale, lines, rho, theta, threshold, minLineLenght, maxLineGap);
+
+        double meanAngle = 0;
+        Log.d(TAG, "rows = " + lines.cols() + "\ncols = " + lines.cols());
+
+        //Analizes the text line per line
+        for (int i = 0; i < lines.rows(); i++) {
+            //Get points from the beginning and the ending of the line of text
+            double[] vec = lines.get(i, 0);
+
+            //First point
+            double x1 = vec[0];
+            double y1 = vec[1];
+
+            //Second point
+            double x2 = vec[2];
+            double y2 = vec[3];
+
+            //Sum all the angle of the lines of text
+            meanAngle += Math.atan2(y2 - y1, x2 - x1);
+        }
+
+        //Calculate the meanAngle by dividing it with the number of rows
+        meanAngle /= lines.rows();
+
+        //Transform the angle in degrees
+        double degreesAngle = Math.toDegrees(meanAngle);
+        Log.i(TAG, "Mean angle=" + degreesAngle);
+        return degreesAngle;
+    }
+
+    /**
+     * Rotate the given bitmap with the given angle
+     * @param original The image that we want to corrected
+     * @param degrees The degrees of the angle we want to rotate
+     * @return the rotatedImage. If the image is null return the original image
+     * @author Thomas Porro(g1), Giovanni Fasan (g1), Oscar Garrido (g1)
+     */
+    private static Bitmap rotateBitmap(Bitmap original, double degrees) {
+
+        //TODO not correct because it returns null, maybe we should add a @NonNull
+        if(original == null){
+            return original;
+        }
+
+        //Obtain the dimen of the image
+        int width = original.getWidth();
+        int height = original.getHeight();
+
+        //Prepare the rotation matrix. The minus before degrees allows us to rotate the
+        //image in the right way
+        Matrix matrix = new Matrix();
+        matrix.preRotate((float)-degrees);
+
+        //Rotate the Bitmap and returns it
+        return Bitmap.createBitmap(original, 0, 0, width, height, matrix, true);
+
+    }
+
+
+    /**
+     * Performs the skew correction
+     * @param image The image that I want to rotate
+     * @return The image corrected
+     * @author Thomas Porro(g1), Giovanni Fasan (g1), Oscar Garrido (g1)
+     */
+    private Bitmap editSkew(Bitmap image){
+        double angle = computeSkew(image);
+        return rotateBitmap(image, angle);
+    }
+
+
+    /**
+     * Detect if the image is bright
+     * @param imageMat the image we want to detect the brightness
+     * @return 0 if image is neither too bright nor too dark,
+     *         1 if image is too bright,
+     *         2 if image is too dark.
```

Comment:
instead of using 0,1,2 you could declare an enumaration such as `private enum Brightness {GOOD, TOO_BRIGHT, TOO_DARK`} (see https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html)

Reply:
We added it, but unfortunately I had loaded the code before this addition